@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -697,6 +692,15 @@ function placePointLabelAtAnchor(
     // Compute text offset from the anchor point
     const textOffset = computePointTextOffset(label, placement, scale, env, tmpTextOffset);
     textOffset.add(screenPosition);
+
+    // Update output screen position.
+    outScreenPosition.set(textOffset.x, textOffset.y, labelState.renderDistance);
+
+    // Check if icon's label was already rejected.
+    if (isRejected) {
+        return PlacementResult.Rejected;
+    }
```

Comment:
With this change, it doens't make sense any more to pass the isRejected parameter to this function or even to placePointLabel.
If the icon is rejected, we can't just skip the call to placePointLabel and reject the text as well.

Reply:
Yet here you're rejecting the text if the icon was rejected without checking whether it's optional or not. Then why not reject it sooner?

There's also other places that wouldn't work for an optional icon:

- placePointLabel is rejecting new labels with rejected icon
https://github.com/heremaps/harp.gl/commit/9584f668f38a121b8881dc9f64052207313cf5cf#diff-cbaa0c98ceb02c4586a463f1982b813f5555d33eab176cd4f180be8eaad25e6cR442-R443

- placePointLabel is checking only the current anchor for permanent labels with rejected icon
https://github.com/heremaps/harp.gl/commit/9584f668f38a121b8881dc9f64052207313cf5cf#diff-cbaa0c98ceb02c4586a463f1982b813f5555d33eab176cd4f180be8eaad25e6cR449-R458

Afaik, MarkerTechnique's iconIsOptional only applies when the icon image is not available for some reason.See:

https://github.com/heremaps/harp.gl/blob/69af4d17600970a9653951e234716da6d1f0ac4b/%40here/harp-datasource-protocol/lib/TechniqueParams.ts#L656
`
/**
     * If true, the text will appear even if the icon cannot be rendered because of missing icon
     * graphics.
     ...
`
That's why currently we only check that property in shouldRenderPointText() if the poiInfo is not valid.

I'm fine changing the behaviour of 'iconIsOptional', actually I think the current behaviour is weird. But for that we can change shouldRenderPointText() to return false if the icon is required but it was rejected. Then I don't see why placePointText needs to know whether the icon was rejected.

Reply:
removed the line.

Reply:
Ok, but we still reject new text with rejected icon:
https://github.com/heremaps/harp.gl/pull/1888/files#diff-cbaa0c98ceb02c4586a463f1982b813f5555d33eab176cd4f180be8eaad25e6cR442-R444

Both new and permanent labels should do the same if the icon is rejected. I suggest to check if the icon is rejected and not optional in shouldRenderPointText() and skip completely the call to placePointLabel. Then placePointLabel() won't need to know if the icon is rejected anymore.

Reply:
Moved test to shouldRenderPointText()

Reply:
... had to move the test back into the calling method, since the case where a rejected icon has to be rendered because it is being faded out would keep the text from being rendered (in which case shouldRenderPointText() would return false).
I think that this should be OK now.